### DISCUSSION TITLE
Random RangeError

### DISCUSSION THREAD

**Discussion by julienbfabre** [google/model-viewer#3383] [Q&A]
Every once in a while we run into the issue of the 3d model not loading in ModelViewer, but if we reload the page the model loads fine and every time afterwards (and every time before also). this is very hard to reproduce so it's very hard to debug, I was hoping that by showing you the error you could either have a clue about what would cause this and maybe either say if this could be an issue of modelviewer or if the problem is definitely somewhere else.

```
Uncaught (in promise) TypeError in body stream
RangeError: attempting to construct out-of-bounds Float32Array on ArrayBuffer                             model-viewer.js:52556:24
     loadAccessor model-viewer.js:49901
THREE.GLTFLoader: Couldn't load texture blob:https:://cldxr.com/0a5551e2-c8aa-4682-a97e-72a840d59dfc      model-viewer.js:50098:11
(... 3 more of those THREE.GLTFLoader)
```

Note that we only use GLBs, no separated texture files so i don't know why it is trying to load a texture from separate urls that don't even exist. But this all works fine 99% of the time which makes it really hard to debug.

In case you need this info, we load modelviewer like this:

```
<script type="module" src="https://unpkg.com/@google/model-viewer/dist/model-viewer.min.js"></script>
<script nomodule src="https://unpkg.com/@google/model-viewer/dist/model-viewer-legacy.js"></script>
```

**Comment by elalish** [comment]
You can skip the legacy version; we don't support that anymore since IE is defunct. The texture looks like a blob, so I don't think it's really a separate file, just another buffer. Out-of-bounds makes it sound like it may be out of GPU memory, which would explain why it's not easy to repro. You might consider converting your glTF files to use KTX2 textures, as these save a lot of GPU RAM.

**Comment by julienbfabre** [comment]
@elalish thanks for your reply. Out of GPU memory seems hard to believe, this happens with small models of just a few MBs both on mobile and on a desktop with a GTX 1080 of 6GB with nothing else eating up that memory.

What else could cause this?

**Reply by elalish** [reply]
Well, I mean it could be a memory leak. Maybe the dev tools profiler will show something useful.
